python/lance/util/duckdb_ext.py

```diff
@@ -0,0 +1,68 @@
+import duckdb
+import os
+import pathlib
+import platform
+import urllib.request
+import zipfile
+
+
+def install_duckdb_extension(version='latest', unsigned=True):
+    """
+    Install the lance duckdb extension
+
+    Parameters
+    ----------
+    version: str, default 'latest'
+        The version of the extension to install
+    unsigned: bool, default True
+        Whether to turn on allow_unsigned_extensions in duckdb
+    """
+    if version == 'latest':
+        version = _get_latest_version('lance', 'lance_duckdb')
+    uri = _get_uri(version)
+    local_path = _download_and_unzip('lance', 'lance_duckdb', uri)
+    con = duckdb.connect(config={"allow_unsigned_extensions": unsigned})
+    con.install_extension(local_path, force_install=True)
+
+
+def _get_uri(version):
+    uname = platform.uname()
+    arch = uname.machine  # arm64, x86_64
+    system = uname.system
+    system = "osx" if system.lower() == "darwin" else system
+    device = _get_device()
+    zip_name = f'lance.duckdb_extension.{system}.{arch}.{device}.zip'
+    uri_root = 'https://eto-public.s3.us-west-2.amazonaws.com/'
+    uri = os.path.join(uri_root, 'artifacts', 'lance', 'lance_duckdb',
+                       version, zip_name)
+    return uri
+
+
+def _get_device():
+    import torch
+    if torch.cuda.is_available():
+        return f"cu{torch.version.cuda.replace('.', '')}"
```

Comment:
Pytorch usually is built with a few cuda versions. do we want to support them all?

Reply:
i think that's too much infra for us to take on right now unless there's a way we can switch between different cuda versions in a docker container or smth?

Reply:
i think that's too much infra for us to take on right now unless there's a way we can switch between different cuda versions in a docker container or smth?